test/kongintegration/expression_router_test.go

```diff
@@ -0,0 +1,155 @@
+package kongintegration
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/kong/deck/file"
+	"github.com/kong/go-kong/kong"
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/require"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/parser/atc"
+	"github.com/kong/kubernetes-ingress-controller/v2/test/internal/helpers"
+	"github.com/kong/kubernetes-ingress-controller/v2/test/kongintegration/containers"
+)
+
+func TestExpressionsRouterMatchers_GenerateValidExpressions(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	kongC := containers.NewKong(ctx, t)
+	kongClient, err := kong.NewClient(lo.ToPtr(kongC.AdminURL(ctx, t)), &http.Client{})
+	require.NoError(t, err)
+
+	httpBinC := containers.NewHTTPBin(ctx, t)
+
+	testCases := []struct {
+		name            string
+		matcher         atc.Matcher
+		matchRequests   []*http.Request
+		unmatchRequests []*http.Request
+	}{
+		{
+			name:    "exact match on path",
+			matcher: atc.NewPredicateHTTPPath(atc.OpEqual, "/foo"),
+			matchRequests: []*http.Request{
+				helpers.MustHTTPRequest(t, "GET", helpers.MustParseURL(t, "http://a.foo.com").String(), "foo", nil),
```

Comment:
```suggestion
				helpers.MustHTTPRequest(t, http.MethodGet, helpers.MustParseURL(t, "http://a.foo.com").String(), "foo", nil),
```
and the same for other methods